pkg/bench/queue_manager_info.go

```diff
@@ -0,0 +1,80 @@
+package bench
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	common_config "github.com/prometheus/common/config"
+	"github.com/prometheus/common/promlog"
+	"github.com/prometheus/prometheus/config"
+	"github.com/timescale/promscale/pkg/remote"
+)
+
+type qmInfo struct {
+	qm        *remote.QueueManager
+	samplesIn *remote.EwmaRate
+	highestTs *remote.MaxTimestamp
+}
+
+func (qmi *qmInfo) run() {
+	ticker := time.NewTicker(shardUpdateDuration)
+	defer ticker.Stop()
+	for range ticker.C {
+		qmi.samplesIn.Tick()
+	}
+}
+
+func getQM(conf *BenchConfig) (*qmInfo, error) {
+	url, err := url.Parse(conf.WriteEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	rwConf := config.DefaultRemoteWriteConfig
+
+	rwConf.URL = &common_config.URL{
+		URL: url,
+	}
+	rwConf.HTTPClientConfig = common_config.HTTPClientConfig{}
+
+	c, err := remote.NewWriteClient("client", &remote.ClientConfig{
+		URL:              rwConf.URL,
+		Timeout:          rwConf.RemoteTimeout,
+		HTTPClientConfig: rwConf.HTTPClientConfig,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	qmi := &qmInfo{
+		samplesIn: remote.NewEwmaRate(ewmaWeight, shardUpdateDuration),
+		highestTs: remote.NewMaxTimestamp(
+			prometheus.NewGauge(prometheus.GaugeOpts{
+				Namespace: "ns",
+				Subsystem: "ss",
+				Name:      "highest_timestamp_in_seconds",
+				Help:      "Highest timestamp that has come into the remote storage via the Appender interface, in seconds since epoch.",
+			})),
+	}
+	qmi.qm = remote.NewQueueManager(
+		remote.NewQueueManagerMetrics(nil, "", ""),
+		nil,
+		nil,
+		promlog.New(&promlog.Config{}),
+		"",
+		qmi.samplesIn,
+		rwConf.QueueConfig,
+		rwConf.MetadataConfig,
+		nil,
```

Comment:
we might wanna attach external labels which would be the benchmark job name, just like what prom-migrator does.